LibreNMS/OS/Timos.php

```diff
@@ -84,6 +84,35 @@ private function nokiaIfName($tmnxPortId, $scheme)
             $port = bindec(substr($map, -21, 6));
             return $slot . "/" . $mda . "/" . $port;
         }
+        if ($scheme == 'schemeB') {
```

Comment:
This function returns garbage for all Ports of a SR-1 (schemeB):
```
   ifIndex ifName               nokiaIfName()       
1090523137 pxc-1.a              0/0/0               
1090523138 pxc-1.b              0/0/0               
1140918273 esat-1/1/1           0/4/0               
1140918274 esat-1/1/2           0/4/0               
1140918275 esat-1/1/3           0/4/0               
1140918276 esat-1/1/4           0/4/0               
1140918277 esat-1/1/5           0/4/0               
1140918278 esat-1/1/6           0/4/0               
1140918279 esat-1/1/7           0/4/0               
1140918280 esat-1/1/8           0/4/1               
1140918281 esat-1/1/9           0/4/1               
1140918282 esat-1/1/10          0/4/1               
1140918332 esat-1/1/60          0/4/7               
1140918333 esat-1/1/61          0/4/7               
1140918334 esat-1/1/62          0/4/7               
1140918335 esat-1/1/63          0/4/7               
1140918336 esat-1/1/64          0/4/8               
1140940928 esat-1/1/c65         0/5/16              
1140941056 esat-1/1/c66         0/5/32              
1140941184 esat-1/1/c67         0/5/48              
1140941312 esat-1/1/c68         0/5/64              
1140973697 esat-1/1/c65/u1      0/7/16              
1526726657 pw-1                 pw1                 
1526759423 pw-32767             pw1023              
1610899520 1/1/c1               1/1/8               
1610899521 1/1/c1/1             1/1/8               
1610899584 1/1/c2               1/1/16              
1610899585 1/1/c2/1             1/1/16              
1610899648 1/1/c3               1/1/24              
1610899712 1/1/c4               1/1/32              
1610899776 1/1/c5               1/1/40              
1610899777 1/1/c5/1             1/1/40              
1610899840 1/1/c6               1/1/48              
1610899841 1/1/c6/1             1/1/48              
```

Why do you try to reverse engineer the Names? $tmnxPortId is the ifIndex, so you could fetch ifName from the Database. That would work for all possible schemes and port types.

Reply:
good catch, I'l try to rework this part.